daemon-tests/Cargo.toml

```diff
@@ -9,10 +9,14 @@ anyhow = "1"
 async-trait = "0.1"
 bdk-ext = { path = "../bdk-ext" }
 daemon = { path = "../daemon" }
+futures = "0.3"
 maker = { path = "../maker" }
 mockall = "0.11"
 mockall_derive = "0.11"
 model = { path = "../model" }
+opentelemetry = { version = "0.17.0", features = ["rt-tokio"] }
+opentelemetry-otlp = { version = "0.10.0", features = ["grpc-sys", "trace"] }
```

Comment:
it feels like one (or both), should really be inside `otel-tests` (you can't use `otel-tests` without it)

Reply:
I can do this, but it'd require two crates:
- otel_tests: exports `otel_test` and 
- otel_tests_macro: defines the macro itself

This is because proc macros, last I checked, couldn't be defined in regular library crates. So, the macro itself needs a separate crate.

Reply:
I've refactored into this model in the latest commit